Jenkinsfile

```diff
@@ -0,0 +1,71 @@
+node ('virtualbox') {
+
+  def directory = "packer-template"
+  env.PACKER_CACHE_DIR = "${env.JENKINS_HOME}/packer_cache"
+  stage 'Clean up'
+  deleteDir()
+
+  stage 'Checkout'
+  sh "mkdir $directory"
+  dir("$directory") {
+    try {
+        checkout scm
+    } catch (e) {
+        currentBuild.result = 'FAILURE'
+        notifyBuild(currentBuild.result)
+        throw e
+    }
+  }
+  dir("$directory") {
+    stage 'bundle'
+    try {
+        sh "bundle install --path ${env.JENKINS_HOME}/vendor/bundle"
+    } catch (e) {
+        currentBuild.result = 'FAILURE'
+        notifyBuild(currentBuild.result)
+        throw e
+    }
+
+    stage 'rake reallyenglish:build'
+    try {
+      sh 'bundle exec rake reallyenglish:test'
+    } catch (e) {
+        currentBuild.result = 'FAILURE'
+        notifyBuild(currentBuild.result)
+        throw e
+    } finally {
+      sh 'bundle exec rake reallyenglish:clean'
```

Comment:
because?

although i do not understand the question, i am going to explain how the tasks was designed.

the requirements were:

* you can build and test a single box by a single command because often you want to add a new version, or update a single box. when you want to multiple boxes at once, either `rake test` or multiple `do_test[host]` in `for` loop can be used
* `do_test` builds and destroys a VM one by one. you do not want multiple VMs running at the same time. if you really want to do so, you can always run commands that the task runs
* `do_test`, and thus `test`, does not remove created box image because if successful, you want to upload to vagrant cloud.
* in Jenkins environment, ensure no box files are left after successful or failed build

does these explanations clarify your question?